packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -861,47 +927,100 @@ const runGenerator = (context, {outputDir, templateVersion, verbose}) => {
+                outputDir: devOutputDir,
+                templateVersion,
+                verbose,
+                isRecursiveCall: true
+            })
+
+            // Update the typescript-minimal dev package.json with dependencies
+            updatePackageJson(p.resolve(devOutputDir, 'package.json'), {
+                devDependencies: {[answers.project.name]: 'file:../'},
+                mobify: {app: {extensions: [answers.project.name]}}
+            })
 
-        // Update the generated project's package.json. NOTE: For bootstrapped projects this
-        // can be done in the template building. But since we have two types of project builds,
-        // (bootstrap/bundle) we'll do it here where it works in both scenarios.
-        const pkgJsonPath = p.resolve(outputDir, 'package.json')
-        const pkgJSON = readJson(pkgJsonPath)
+            // Create the .npmignore file, excluding the typescript-minimal dev folder
+            createNpmIgnoreFile(outputDir, [`${APP_EXTENSIBILITY_TYPESCRIPT_MINIMAL_DEV_DIR}/`])
+
+            //TODO: Avoid duplicated console.log Installing dependencies in terminal
+            npmInstall(devOutputDir, {verbose})
+        } else {
+            processAppExtensions(selectedAppExtensions, extractAppExtensions, appExtensionsDir)
+        }
 
         // Add selected Application Extensions to devDependencies and mobify object
         const appExtensionDeps = selectedAppExtensions.reduce((acc, appExtensionName) => {
             // Find the corresponding Application Extension details
-            const appExtension = context.appExtensions.find((ext) => ext.value === appExtensionName)
-            const version = appExtension ? appExtension.version : '1.0.0-dev'
+            const appExtensionDetails = context.availableAppExtensions.find(
+                (ext) => ext.value === appExtensionName
+            )
+            const version = appExtensionDetails ? appExtensionDetails.version : '1.0.0-dev'
 
             acc[appExtensionName] = extractAppExtensions
                 ? `file:./app/application-extensions/${appExtensionName}`
                 : version
             return acc
         }, {})
 
-        const finalPkgData = merge(pkgJSON, {
-            name: slugifyName(context.answers.project.name || context.preset.id),
+        updatePackageJson(p.resolve(outputDir, 'package.json'), {
+            name: (() => {
+                const projectName = context.answers.project.name || context.preset.id
+                // Match @namespace/project-name pattern
+                const namespaceRegex = /^(@[a-zA-Z0-9-_]+)\/(.+)$/
+                const match = projectName.match(namespaceRegex)
+
+                if (match) {
+                    const [, namespace, actualName] = match
+                    // Slugify only the project name part
+                    return `${namespace}/${slugifyName(actualName)}`
+                }
+
+                return slugifyName(projectName)
```

Comment:
Can you give me a brief summary of what the change is here. At a glance it looks like this is being run for both template projects and app extension projects, but we've introduced some parsing of the package name and removing the namespace?

Reply:
Since now the project name includes the namespace (e.g. @namespace/project-name), the slugify function was removing the special characters from the namespace. The change is to only slugify the project name part to avoid removing the '@' and '/' symbols during the slugify process.